File: app/src/main/java/com/example/patiobalmax/database/PuestoDao.java
package com.example.patiobalmax.database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface PuestoDao {

    @Insert
    void insert(PuestoEntity puesto);

    @Update
    void update(PuestoEntity puesto);

    @Delete
    void delete(PuestoEntity puesto);

    @Query("SELECT * FROM puestos")
    LiveData<List<PuestoEntity>> getAll();

    @Query("SELECT * FROM puestos WHERE id_patio = :idPatio")
    LiveData<List<PuestoEntity>> getAllByPatioId(int idPatio);
}

File: app/src/main/kotlin/com/example/patiobalmax/database/dao/PuestoDao.kt
package com.example.patiobalmax.database.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import com.example.patiobalmax.database.entity.Puesto

@Dao
interface PuestoDao {

    @Insert
    suspend fun insert(puesto: Puesto)

    @Update
    suspend fun update(placa1: String, placa2: String, puestoId: Int)

    @Query("UPDATE puestos SET placa1 = :placa1, placa2 = :placa2 WHERE id = :puestoId")
    suspend fun actualizarPatentes(placa1: String, placa2: String, puestoId: Int)

    @Query("SELECT * FROM puestos WHERE patioId = :patioId")
    suspend fun getByPatio(patioId: Int): List<Puesto>

    @Query("SELECT * FROM puestos WHERE id = :puestoId")
    suspend fun getById(puestoId: Int): Puesto?
}
